mlir/lib/Dialect/Rock/Transforms/AlignTiling.cpp

```diff
@@ -96,24 +96,29 @@ static Value transformAccordingly(PatternRewriter &b, Value nOut, Value refOp) {
     nOut = transformAccordingly(b, nOut, rtop.getOperand());
 
     // 1. apply identical transforms to other inputs
-    BlockAndValueMapping cmap;
-    cmap.map(rtop.getOperand(), nOut);
-    auto ntop = dyn_cast<TransformOp>(b.clone(*rtop, cmap));
-    nOut = ntop.getResult();
+    nOut = b.create<TransformOp>(rtop.getLoc(), nOut, rtop.getTransform());
   }
 
   return nOut;
 }
 
+Value makeRegs(PatternRewriter &b, MemRefType::Builder &mrb, Location loc,
+               Type srcType) {
+  auto srcMemType = srcType.cast<MemRefType>();
+  // 1. create a second allocation of the same type to hold loaded elements
+  return b.create<GpuAllocOp>(loc, static_cast<MemRefType>(mrb.setElementType(
```

Comment:
Address space needs to be set here.

Reply:
It gets that from the MemRefType::Builder which is initialized by the register type for the blockwise_gemm output.